File: HealthcareSystem.Application/Requests/AuthRequests.cs
namespace HealthcareSystem.Application.Requests;

public static class AuthRequests {
    public record RegisterRequest(
        string Username, string Email, string Gender,
        string Password, string ConfirmPassword, bool EnableTwoFactor
    );

    public record LoginRequest(string Email, string Password);

    public record TwoFactorRequest(string Email, string Token, string Provider);

    public record RefreshTokenRequest(string AccessToken, string RefreshToken);

    public record ForgetPasswordRequest(string Email, string ClientUri);

    public record ResetPasswordRequest(
        string Email, string Token, string Password, string ConfirmPassword
    );
}

File: HealthcareSystem.Client/src/app/data/requests/authRequests.ts
export interface RegisterRequest {
  username: string;
  email: string;
  gender: string;
  password: string;
  confirmPassword: string;
  enableTwoFactor: boolean;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface TwoFactorRequest {
  email: string;
  token: string;
  provider: string;
}

export interface RefreshTokenRequest {
  accessToken: string;
  refreshToken: string;
}

export interface ResetPasswordRequest {
  email: string;
  token: string;
  password: string;
  confirmPassword: string;
}
